packages/venia-concept/src/actions/cart.js

```diff
@@ -1,42 +1,61 @@
+import { createActions } from 'redux-actions';
 import { RestApi } from '@magento/peregrine';
 
 import { closeDrawer, toggleDrawer } from 'src/actions/app';
+import checkoutActions from 'src/actions/checkout';
+import BrowserPersistence from 'src/util/simplePersistence';
+
+const prefix = 'CART';
+const actionTypes = [
+    'ADD_ITEM',
+    'REQUEST_GUEST_CART',
+    'RECEIVE_GUEST_CART',
+    'REQUEST_DETAILS',
+    'UPDATE_DETAILS'
+];
+
+const actions = createActions(...actionTypes, { prefix });
+export default actions;
+
+/* async action creators */
 
 const { request } = RestApi.Magento2;
+const storage = new BrowserPersistence();
```

Comment:
Since actions are similar to events, a reducer can seem like a good place to "listen" for "events" and perform side effects such as local persistence. In fact, though, one of the few firm rules in Redux says that [reducers should perform no side effects](https://redux.js.org/basics/reducers).

> Things you should **never** do inside a reducer:
>
> - Mutate its arguments;
> - Perform side effects like API calls and routing transitions;
> - Call non-pure functions, e.g. `Date.now()` or `Math.random()`.

So I had to migrate the `storage` logic from the cart reducer here to the cart actions. It's largely unchanged. However, an alternate way to handle this logic on the "listening" side would be to put it in a middleware.